recipes/libcurl/all/conanfile.py

```diff
@@ -285,9 +304,53 @@ def _configure_autotools_vars(self):
 
             del autotools_vars['LIBS']
             self.output.info("Autotools env vars: " + repr(autotools_vars))
+
+        if tools.cross_building(self.settings):
+            if self.settings.os == "iOS": 
+                ios_dev_target = str(self.settings.os.version).split(".")[0]
+                if self.settings.arch in ["x86", "x86_64"]:
+                    iphoneos = "iphonesimulator"
+                    autotools_vars['CPPFLAGS'] = "-D__IPHONE_OS_VERSION_MIN_REQUIRED={}0000".format(ios_dev_target)
+                elif self.settings.arch in ["armv7", "armv7s", "armv8"]:
+                    iphoneos = "iphoneos"
+                    autotools_vars['CPPFLAGS'] = ""
+                else:
+                    raise ConanInvalidConfiguration("Unsuported iOS arch {}".format(self.settings.arch)) 
+
+                cc = tools.XCRun(self.settings, iphoneos).cc
+                sysroot = "-isysroot {}".format(tools.XCRun(self.settings, iphoneos).sdk_path)
+
+                if self.settings.arch == "armv8":
+                    configure_arch = "arm64"    
+                    configure_host = "arm" #unused, autodetected
+                else:
+                    configure_arch = self.settings.arch
+                    configure_host = self.settings.arch #unused, autodetected
+    
+
+                arch_flag = "-arch {}".format(configure_arch)
+                ios_min_version = "-miphoneos-version-min={}".format(ios_dev_target)
+                bitcode =  "-fembed-bitcode" if self.options.enable_bitcode else ""
+                extra_flag = "-Werror=partial-availability"
+                extra_def = " -DHAVE_SOCKET -DHAVE_FCNTL_O_NONBLOCK"
+                # if we debug, maybe add a -gdwarf-2 , but why would we want that?  
+
+                autotools_vars['CC'] = cc 
+                autotools_vars['IPHONEOS_DEPLOYMENT_TARGET'] = ios_dev_target
+                autotools_vars['CFLAGS'] = "{} {} {} {} {}".format(
+                    sysroot, arch_flag, ios_min_version, bitcode, extra_flag
+                ) 
+                autotools_vars['LDFLAGS'] = "{} {}".format(arch_flag, sysroot)
+                autotools_vars['CPPFLAGS'] += extra_def
+
+            elif self.settings.os == "Android":
```

Comment:
same, pls avoid adding empty code blocks

Reply:
shall I also remove the `system_requirements`. function?

Reply:
yes, if it does nothing

Reply:
if there is documentation, like, it should be like this, or hey, for Android, its tested but there is nothing to do, I think I prefer having this in the file, since it is documentation

But I removed the parts that did nothing and had no docu purpose